crypto/bignum.c

```diff
@@ -1781,10 +1781,17 @@ size_t bn_format(const bignum256 *amount, const char *prefix, const char *suffix
     // decimals == 0 && exponent == 0
 
     // Add significant digits
+    int digits = 0;
+    bool is_zero = false;
     do {
+      ++digits;
       bn_divmod10(&temp, &digit);
+      is_zero = bn_is_zero(&temp);
       BN_FORMAT_ADD_OUTPUT_CHAR('0' + digit)
-    } while (!bn_is_zero(&temp));
+      if (!is_zero && digits % 3 == 0) {
+        BN_FORMAT_ADD_OUTPUT_CHAR(',')
+      }
+    } while (!is_zero);
```

Comment:
I think this won't work correctly if `exponent > decimals`. We need to start counting the digits and adding separators in the block [above](https://github.com/trezor/trezor-firmware/pull/2425/files#diff-ae29a9bad321bf657961dd96f640a325907b9b1878fa746179e27909caefb95fR1777-R1779).

Reply:
fixed in [11db39c](https://github.com/trezor/trezor-firmware/pull/2425/commits/11db39c3720a03b4ac8b75b4aa4551a65d3c4fb3)